packages/token/src/index.ts

```diff
@@ -134,12 +159,40 @@ export class Token {
         return formatUnits(amount, this.decimals);
     }
 
+    /**
+     * Get a token from a ticker on a specific chain.
+     *
+     * @param ticker The ticker of the token to get.
+     * @param chain The chain to get the token from.
+     */
+    static fromTickerOnChain(ticker: string, chain: ChainId): Token {
+        if (!Token.tokenMappings[chain]) {
+            throw new Error(
+                `Token mapping for chain ${chain} not initialized. Call fetchRemapFromRepo first.`,
+            );
+        }
+
+        const tokenData = Token.tokenMappings[chain].tokens.find(
+            (token) => token.ticker === ticker,
+        );
+        if (!tokenData) {
+            return DEFAULT_TOKEN;
+        }
+
+        return new Token(tokenData);
+    }
+
+    /** @deprecated Use `Token.fromTickerOnChain` instead */
     static fromTicker(ticker: string): Token {
-        if (Token.tokenMapping.tokens.length === 0) {
-            throw new Error("Token not initialized. Call fetchRemapFromRepo first.");
+        if (!Token.tokenMappings.default) {
+            throw new Error(
+                "Default token mapping not initialized. Call fetchRemapFromRepo first.",
+            );
```

Comment:
A helper that either returns chain or errors would be nice for all the places we guard.